jsonparse.js

```diff
@@ -408,6 +415,34 @@ proto.onToken = function (token, value) {
   }
 };
 
+/**
+ * determines if precision is lost when parsing the JSON representation of a number to a javascript number
+ * @param {string} jsonNumber
+ * @param {number} javascriptNumber
+ * @return {boolean}
+ */
+function precisionWasLost (jsonNumber, javascriptNumber) {
+  return getDigitsSpecifyingPrecision(jsonNumber) !== getDigitsSpecifyingPrecision(JSON.stringify(javascriptNumber))
+}
+
+/**
+ * Takes in a string represention of a number and returns a simplified version of the significand
+ * as a string.
+ *
+ * The simplified significand only contains the digits and we remove any trailing zeros or decimal
+ * information.
+ * @param {string} numberString
+ * @return {string}
+ */
+function getDigitsSpecifyingPrecision(numberString) {
+  return numberString
+    .replace(/\./, '') // drops the dot if it exists
+    .replace(/^-?0?/, '') // drops leading negative sign and and leading 0's if present
```

Comment:
Should we drop leading zeroes _before_ dropping the decimal point?

For example, `'.0012'` would currently become `'12'`, but I think we want it to be `'0012'` because leading zeroes _after_ the decimal point matter, right?

Reply:
I think we want it to become `12` in that case. Leading zeroes are arbitrary, and there can be any number of them depending what power is in the exponent. For example, 12 is the same as 0.012e3.